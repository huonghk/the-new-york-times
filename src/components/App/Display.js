import React from 'react';
import PropTypes from 'prop-types';
import Article from 'components/Article';
import Pagination from 'components/Pagination';
import styles from './styles.module.css';

const AppDisplay = ({ articles }) => {
  let articlesRef = React.createRef();
  return (
    <React.Fragment>
      <header className={styles.header}>
        <div className="container">
          <h2>The New York Times</h2>
        </div>
      </header>
      <main className={styles.body} ref={articlesRef}>
        <div className="container">
          <div>
            {articles.map(item => (
              <div className={styles.item} key={item.id}>
                <Article {...item} />
              </div>
            ))}
          </div>
          <div className={styles.pagination}>
            <Pagination articlesRef={articlesRef} />
          </div>
        </div>
      </main>
      <footer className={styles.footer}>
        <div className="container">
          Â© 2018 The New York Times Company
        </div>
      </footer>
    </React.Fragment>
  );
};

AppDisplay.propTypes = {
  articles: PropTypes.array.isRequired,
};

export default AppDisplay;
