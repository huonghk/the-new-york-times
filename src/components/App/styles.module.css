.header {
  padding: 32px 0;
  border-bottom: 4px solid #262626;
  position: relative;
  background: rgba(0,0,0,0.04);
  text-transform: uppercase;
}

.footer {
  padding: 32px 0;
  border-top: 4px solid #262626;
  position: relative;
  background: rgba(0,0,0,0.04);
  color: #262626;
  text-align: center;
}

.header:after,
.footer:after {
  content: '';
  height: 1px;
  width: 100%;
  background: #000000;
  position: absolute;
  left: 0;
  right: 0;
}
.header:after {
  bottom: -6px;
}
.footer:after {
  top: -6px;
}

.body {
  padding: 48px 0;
}
.item {
  padding-bottom: 24px;
  margin-bottom: 24px;
  border-bottom: 1px dotted rgba(55,55,55,0.2);
}
