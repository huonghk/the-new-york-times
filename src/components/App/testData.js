export const articles = [
  {
    id: "5b676a59068401528a2d28f4",
    multimedia: {
      small: "https://www.nytimes.com/images/2018/08/06/world/06singapore-china-1/00singapore-china-1-smallSquare252.jpg",
      large: "https://www.nytimes.com/images/2018/08/06/world/06singapore-china-1/merlin_138550440_6f0db81d-20dc-43d7-a8bc-ac028ea45bc7-popup.jpg",
    },
    pubDate: "2018-08-05T21:21:27+00:00",
    snippet:  "With ethnic Chinese constituting a majority of Singapore’s population, some fear the country could be an especially tantalizing target for China’s influence.",
    source: "The New York Times",
  },
  {
    id: "5b1a5329068401528a2b23d4",
    multimedia: {
      small: "https://www.nytimes.com/images/2018/06/09/world/09singapore-2-print/09singapore-1-smallSquare252.jpg",
      large: "https://www.nytimes.com/images/2018/06/09/world/09singapore-2-print/merlin_139234518_b706ad2a-2077-4376-b6d2-a2fd8412ab4d-popup.jpg",
    },
    pubDate: "2018-06-08T09:57:59+00:00",
    snippet:  "The lavish hotel where President Trump and Kim Jong-un will meet is surrounded by peacocks, golf resorts and a security zone that bars all trappings of protest.",
    source: "The New York Times",
  },
];
