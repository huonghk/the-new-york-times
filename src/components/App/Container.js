import React, { Component } from 'react';
import PropTypes from 'prop-types';
import Display from './Display';

class AppContainer extends Component {
  componentDidMount() {
    this.props.loadArticles();
  }

  render() {
    const { loadArticles, ...props } = this.props;
    return (
      <Display {...props} />
    );
  }
}

AppContainer.propTypes = {
  loadArticles: PropTypes.func.isRequired,
};

export default AppContainer;
