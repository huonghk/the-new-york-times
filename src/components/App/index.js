import { connect } from 'react-redux';
import actions from 'store/article/actions';
import Container from './Container';

const mapStateToProps = state => ({
  loading: state.article.loading,
  articles: state.article.data,
});

const mapDispatchToProps = dispatch => ({
  loadArticles: () => dispatch(actions.load(1)),
});

export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(Container);
