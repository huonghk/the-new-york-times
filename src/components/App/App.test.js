import React from 'react';
import { articles } from './testData';
import { shallow } from 'enzyme';
import App from './Container';

describe('App Component', () => {
  let wrapper;
  const mockloadArticles = jest.fn();
  beforeEach(() => {
    mockloadArticles.mockClear();
    // pass the mock function as the login prop
    wrapper = shallow(
      <App
        articles={articles}
        loadArticles={mockloadArticles}
      />,
    );
  });

  it('renders without crashing', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should call the mock loadArticles function on Mount', () => {
    expect(mockloadArticles.mock.calls.length).toBe(1);
  });
});
