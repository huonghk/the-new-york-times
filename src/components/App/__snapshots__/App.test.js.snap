// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component renders without crashing 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <AppContainer
    articles={
      Array [
        Object {
          "id": "5b676a59068401528a2d28f4",
          "multimedia": Object {
            "large": "https://www.nytimes.com/images/2018/08/06/world/06singapore-china-1/merlin_138550440_6f0db81d-20dc-43d7-a8bc-ac028ea45bc7-popup.jpg",
            "small": "https://www.nytimes.com/images/2018/08/06/world/06singapore-china-1/00singapore-china-1-smallSquare252.jpg",
          },
          "pubDate": "2018-08-05T21:21:27+00:00",
          "snippet": "With ethnic Chinese constituting a majority of Singapore’s population, some fear the country could be an especially tantalizing target for China’s influence.",
          "source": "The New York Times",
        },
        Object {
          "id": "5b1a5329068401528a2b23d4",
          "multimedia": Object {
            "large": "https://www.nytimes.com/images/2018/06/09/world/09singapore-2-print/merlin_139234518_b706ad2a-2077-4376-b6d2-a2fd8412ab4d-popup.jpg",
            "small": "https://www.nytimes.com/images/2018/06/09/world/09singapore-2-print/09singapore-1-smallSquare252.jpg",
          },
          "pubDate": "2018-06-08T09:57:59+00:00",
          "snippet": "The lavish hotel where President Trump and Kim Jong-un will meet is surrounded by peacocks, golf resorts and a security zone that bars all trappings of protest.",
          "source": "The New York Times",
        },
      ]
    }
    loadArticles={
      [MockFunction] {
        "calls": Array [
          Array [],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": undefined,
          },
        ],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "articles": Array [
        Object {
          "id": "5b676a59068401528a2d28f4",
          "multimedia": Object {
            "large": "https://www.nytimes.com/images/2018/08/06/world/06singapore-china-1/merlin_138550440_6f0db81d-20dc-43d7-a8bc-ac028ea45bc7-popup.jpg",
            "small": "https://www.nytimes.com/images/2018/08/06/world/06singapore-china-1/00singapore-china-1-smallSquare252.jpg",
          },
          "pubDate": "2018-08-05T21:21:27+00:00",
          "snippet": "With ethnic Chinese constituting a majority of Singapore’s population, some fear the country could be an especially tantalizing target for China’s influence.",
          "source": "The New York Times",
        },
        Object {
          "id": "5b1a5329068401528a2b23d4",
          "multimedia": Object {
            "large": "https://www.nytimes.com/images/2018/06/09/world/09singapore-2-print/merlin_139234518_b706ad2a-2077-4376-b6d2-a2fd8412ab4d-popup.jpg",
            "small": "https://www.nytimes.com/images/2018/06/09/world/09singapore-2-print/09singapore-1-smallSquare252.jpg",
          },
          "pubDate": "2018-06-08T09:57:59+00:00",
          "snippet": "The lavish hotel where President Trump and Kim Jong-un will meet is surrounded by peacocks, golf resorts and a security zone that bars all trappings of protest.",
          "source": "The New York Times",
        },
      ],
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "articles": Array [
          Object {
            "id": "5b676a59068401528a2d28f4",
            "multimedia": Object {
              "large": "https://www.nytimes.com/images/2018/08/06/world/06singapore-china-1/merlin_138550440_6f0db81d-20dc-43d7-a8bc-ac028ea45bc7-popup.jpg",
              "small": "https://www.nytimes.com/images/2018/08/06/world/06singapore-china-1/00singapore-china-1-smallSquare252.jpg",
            },
            "pubDate": "2018-08-05T21:21:27+00:00",
            "snippet": "With ethnic Chinese constituting a majority of Singapore’s population, some fear the country could be an especially tantalizing target for China’s influence.",
            "source": "The New York Times",
          },
          Object {
            "id": "5b1a5329068401528a2b23d4",
            "multimedia": Object {
              "large": "https://www.nytimes.com/images/2018/06/09/world/09singapore-2-print/merlin_139234518_b706ad2a-2077-4376-b6d2-a2fd8412ab4d-popup.jpg",
              "small": "https://www.nytimes.com/images/2018/06/09/world/09singapore-2-print/09singapore-1-smallSquare252.jpg",
            },
            "pubDate": "2018-06-08T09:57:59+00:00",
            "snippet": "The lavish hotel where President Trump and Kim Jong-un will meet is surrounded by peacocks, golf resorts and a security zone that bars all trappings of protest.",
            "source": "The New York Times",
          },
        ],
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
